Jenkinsfile

```diff
@@ -2,38 +2,194 @@
+                                    ssh -v -o StrictHostKeyChecking=no -i $FILE -NL 5556:127.0.0.1:5555 root@$IP &
+                                    $ANDROID_HOME/platform-tools/adb connect 127.0.0.1:5556
+                                    $ANDROID_HOME/platform-tools/adb wait-for-device
+                                '''
+                            }
+                        }
+                        stage('Run Tests') {
+                            timeout(60) {
+                                sh'''#!/bin/bash -l
+                                    set -x
+                                    set -e
+                                    appium &
+                                    sleep 10
+                                    export app=$PWD/mozilla-release/objdir-frontend-android/dist/$APP
+                                    cd autobots
+                                    virtualenv ~/venv
+                                    source ~/venv/bin/activate
+                                    chmod 0755 requirements.txt
+                                    pip install -r requirements.txt
+                                    python testRunner.py || true
+                               '''
+                           }
+                        }
+                    }
+                    try {
+                        stage('Upload APK') {
+                            archiveArtifacts allowEmptyArchive: true, artifacts: 'mozilla-release/objdir-frontend-android/dist/*.apk'
+                            archiveArtifacts allowEmptyArchive: true, artifacts: 'autobots/*.log'
+                            junit 'autobots/test-reports/*.xml'
+                            zip archive: true, dir: 'autobots/screenshots', glob: '', zipFile: 'autobots/screenshots.zip'
+                        }
+                    } catch(e) {
+                        print e
+                    }
+                }
             }
-        }
-        finally {
+        } finally {
             stage('Clean Up') {
-                sh '''#!/bin/bash -l
+                sh '''#!/bin/bash
                     rm -f mozilla-release/mozconfig
                     rm -rf mozilla-release/objdir-frontend-android
+                    rm -rf autobots
                 '''
             }
         }
     }
 }
+
+@NonCPS
+def withGenymotion(Closure body) {
+    try {
+        def instance_id = sh(returnStdout: true, script: '''
+            aws ec2 run-instances \
+            --image-id ami-13050368 \
+            --count 1 \
+            --instance-type t2.medium \
+            --key-name android_ci_genymotion \
+            --security-group-ids sg-5bbf173f \
+            --subnet-id subnet-341ff61f  \
+            --region=us-east-1 \
+            --query "Instances[].InstanceId" \
+            --output text
+            ''').trim()
+        def ip = sh(returnStdout: true, script: """
+            aws ec2 describe-instances \
+            --instance-ids $instance_id \
```

Comment:
I like to add curly braces around groovy variables so they are easier to distinguish from environment variables

Reply:
Yea.. I missed a couple of them. Added now. Done.